src/app/directives/creation-date.directive.ts

```diff
@@ -0,0 +1,30 @@
+import { Directive, Input, ElementRef, Renderer2 } from '@angular/core';
+import { CourseItemInterface } from '../course-list/models/course-item.model';
+
+@Directive({
+  selector: '[appCreationDate]'
+})
+export class CreationDateDirective {
+
+  constructor(private el: ElementRef, private render: Renderer2) { }
+
+  @Input('appCreationDate') set appCreationDate(courseItem: CourseItemInterface) {
```

Comment:
This directive can be more _reusable_ if would receive a simple `Date` (date string) as an Input rather than `CourseItemInterface`